Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperation.swift

```diff
@@ -335,74 +456,102 @@ struct UpdateMany: TestOperation {
         }
-        let result = try collection.deleteMany(self.filter, options: self.options, session: session)
+        let result =
+            try collection.deleteMany(self.filter, options: self.options, session: sessions[self.session ?? ""])
         return TestOperationResult(from: result)
     }
 }
 
 struct DeleteOne: TestOperation {
+    let session: String?
     let filter: Document
     let options: DeleteOptions
 
-    private enum CodingKeys: String, CodingKey { case filter }
+    private enum CodingKeys: String, CodingKey { case session, filter }
 
     init(from decoder: Decoder) throws {
         self.options = try DeleteOptions(from: decoder)
         let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = try container.decodeIfPresent(String.self, forKey: .session)
         self.filter = try container.decode(Document.self, forKey: .filter)
     }
 
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    func execute(on target: TestOperationTarget, sessions: [String: ClientSession]) throws -> TestOperationResult? {
         guard case let .collection(collection) = target else {
             throw TestError(message: "collection not provided to deleteOne")
         }
-        let result = try collection.deleteOne(self.filter, options: self.options, session: session)
+        let result = try collection.deleteOne(self.filter, options: self.options, session: sessions[self.session ?? ""])
         return TestOperationResult(from: result)
     }
 }
 
 struct InsertOne: TestOperation {
+    let session: String?
     let document: Document
 
-    func execute(on target: TestOperationTarget, session _: ClientSession?) throws -> TestOperationResult? {
+    private enum CodingKeys: String, CodingKey { case session, document }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = try container.decodeIfPresent(String.self, forKey: .session)
+        self.document = try container.decode(Document.self, forKey: .document)
+    }
+
+    func execute(on target: TestOperationTarget, sessions: [String: ClientSession]) throws -> TestOperationResult? {
         guard case let .collection(collection) = target else {
             throw TestError(message: "collection not provided to insertOne")
         }
-        return TestOperationResult(from: try collection.insertOne(self.document))
+        return TestOperationResult(from: try collection.insertOne(self.document, session: sessions[self.session ?? ""]))
     }
 }
 
 struct InsertMany: TestOperation {
+    let session: String?
     let documents: [Document]
     let options: InsertManyOptions
 
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    private enum CodingKeys: String, CodingKey { case session, documents }
+
+    init(from decoder: Decoder) throws {
```

Comment:
Despite having an options struct, this one doesn't actually need any special handling. If you look at the retryable-writes form of this operation, it has an `options` field:
```json
"operation": {
    "name": "insertMany",
    "arguments": {
        "documents": [],
        "options": {
            "ordered": true
        }
    }
}
```
In the transactions tests, it has a session field at the same level `options` would be:
```json
{
          "name": "insertMany",
          "object": "collection",
          "arguments": {
            "documents": [],
            "session": "session1"
          },
}
```
So they're compatible, and we can delete this initializer and just rely on the synthesized one.

Reply:
Done.